src/tsickle.ts

```diff
@@ -283,7 +283,10 @@ abstract class ClosureRewriter extends Rewriter {
       }
       // Merge the parameters into a single list of merged names and list of types
       const sig = typeChecker.getSignatureFromDeclaration(fnDecl);
-      if (!sig) throw new Error(`invalid signature ${fnDecl.name}`);
+      if (!sig || !sig.declaration) throw new Error(`invalid signature ${fnDecl.name}`);
+      if (sig.declaration.kind === ts.SyntaxKind.JSDocSignature) {
+        throw new Error(`JSDoc signature ${fnDecl.name}`);
+      }
```

Comment:
Should we be using this for all our JSDoc parsing now?  I am confused about the circumstances where TS obeys JSDoc in a TS file.

Reply:
Not sure. The `JSDocSignature` happens AFAIU when TypeScript resolves an import from TypeScript to a function in JS that has JSDoc declared. We'll probably deal with that more in the future, can revisit then.